tests/tracer/test_rand.py

```diff
@@ -191,17 +202,26 @@ def test_threadsafe():
     assert len(ids) > 0
 
 
+@pytest.mark.subprocess
 def test_tracer_usage_fork():
+    from itertools import chain
+    import os
+
+    from ddtrace import tracer
+    from tests.tracer.test_rand import MPQueue
+
     q = MPQueue()
     pid = os.fork()
 
+    def get_ids():
+        with tracer.start_span("s") as s:
+            yield s.span_id, s.trace_id
+
     # Similar test to test_fork() above except we use the tracer API.
     # In this case we expect to never have collisions.
     if pid > 0:
         # parent
-        parent_ids_list = list(
-            chain.from_iterable((s.span_id, s.trace_id) for s in [tracer.start_span("s") for _ in range(100)])
-        )
+        parent_ids_list = list(chain.from_iterable(ids for ids in [get_ids() for _ in range(100)]))
```

Comment:
This is just to ensure span.finish() on each of these?

Reply:
Yup